pkg/webservice/handlers.go

```diff
@@ -418,6 +420,80 @@ func getNodesUtilJSON(partition *scheduler.PartitionContext, name string) *dao.N
 	}
 }
 
+func getNodeUtilisations(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+	var result []*dao.PartitionNodesUtilDAOInfo
+	for _, part := range schedulerContext.GetPartitionMapClone() {
+		partitionNodesUtilJSON := getPartitionNodesUtilJSON(part)
+		result = append(result, partitionNodesUtilJSON)
+	}
+
+	if err := json.NewEncoder(w).Encode(result); err != nil {
+		buildJSONErrorResponse(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+// getPartitionNodesUtilJSON retrieves the utilization of all resource types for nodes within a specific partition.
+func getPartitionNodesUtilJSON(partition *scheduler.PartitionContext) *dao.PartitionNodesUtilDAOInfo {
+	type UtilizationBucket struct {
+		NodeCount []int      // 10 buckets, each bucket contains number of nodes
+		NodeList  [][]string // 10 buckets, each bucket contains node name list
+	}
+	resourceBuckets := make(map[string]*UtilizationBucket) // key is resource type, value is UtilizationBucket
+
+	// put nodes to buckets
+	for _, node := range partition.GetNodes() {
```

Comment:
just curios. IIRC, the order of `partition.GetNodes` is not specified (as it is based on map). Could it be a potential issue?

Reply:
Apart from nodes, the resource types are also unordered, but I believe this is acceptible. 
There shouldn't be a specific order for node names and resource types.

(One possible ordering is to sort them after calculating the node/resource loading, but I think this is unnecessary. 
The ordering could be controlled in client side.)